synapse/http/client.py

```diff
@@ -388,8 +390,8 @@ def __init__(
             # by the DNS resolution.
             self.agent = BlacklistingAgentWrapper(
                 self.agent,
-                ip_whitelist=self._ip_whitelist,
                 ip_blacklist=self._ip_blacklist,
+                ip_whitelist=self._ip_whitelist,
```

Comment:
I rearranged it in the `BlacklistingAgentWrapper.__init__` since `ip_blacklist` is now required. The callers don't have to match since they use kwargs, but I find it cleaner.